### PR TITLE
Add support for Nix openssl package on macOS

### PR DESCRIPTION
**Pull request by anmonteiro**
Nix can also be used as a package manager in macOS

### CODE REVIEW

**Comment by mseri** [comment]
This requires also updating the hashes in the opam file

**Comment by anmonteiro** [comment]
Ah wasn't aware. Done, thanks

File: packages/conf-openssl/conf-openssl.1/files/osx-build.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/bin/sh
 
-if [ -d "/usr/local/opt/openssl/" ]; then
+if [ -d "$HOME/.nix-profile/lib/pkgconfig/" ]; then
+  # Nix on macOS
+  PKG_CONFIG_PATH=$HOME/.nix-profile/lib/pkgconfig
```
Comment (mseri) [inline packages/conf-openssl/conf-openssl.1/files/osx-build.sh]: Sorry, I just noticed this: I think you need to add ` pkg-config openssl` to this line

Comment (anmonteiro) [reply packages/conf-openssl/conf-openssl.1/files/osx-build.sh]: oh wow my bad, nice catch. Will do right now

Comment (anmonteiro) [reply packages/conf-openssl/conf-openssl.1/files/osx-build.sh]: Done and updated the hash too

File: packages/conf-openssl/conf-openssl.1/files/osx-build.sh [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #!/bin/sh
 
-if [ -d "/usr/local/opt/openssl/" ]; then
+if [ -d "$HOME/.nix-profile/lib/pkgconfig/" ]; then
+  # Nix on macOS
+  PKG_CONFIG_PATH=$HOME/.nix-profile/lib/pkgconfig openssl
```
Comment (mseri) [inline packages/conf-openssl/conf-openssl.1/files/osx-build.sh]: Still not it, I think the correct line should be
```suggestion
  PKG_CONFIG_PATH=$HOME/.nix-profile/lib/pkgconfig pkg-config openssl
```

**Comment by mseri** [comment]
Thanks for the patience

**Comment by pelzlpj** [comment]
👎 

This is actually a regression on my nix config. osx-build.sh seems to assume that `openssl` is installed via "nix-env -i". This breaks development environment isolation; the recommended path is to pull in libraries like `openssl` via a custom `nix-shell` invocation.

**Comment by mseri** [comment]
@pelzlpj would you mind opening a PR to revert this change and add a postinstall message in case of failure that explains what to do if the user uses nix?

**Comment by anmonteiro** [comment]
@pelzlpj Apologies, that is indeed the case. I'm quite new to nix and clearly didn't think there would be other workflows.

**Pull request merged**